File: tp-arithmetic-java/src/test/java/fr/isima/service/OperationCounterTest.java
package fr.isima.service;


import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class OperationCounterTest {

    private OperationCounter operationCounter = new OperationCounter();

    @Test
    public void countRequest_maths_returnsToStringWithMathIncrementedByOne() throws Exception {
        assertThat(operationCounter.countRequest("/maths")).isEqualTo("Math ops : 1 and string ops 0");
    }

    @Test
    public void countRequest_strings_returnsToStringWithStringsIncrementedByOne() throws Exception {
        assertThat(operationCounter.countRequest("/strings")).isEqualTo("Math ops : 0 and string ops 1");
    }

    @Test(expected = IllegalArgumentException.class)
    public void countRequest_noBoundedPath_thorws() throws Exception {
        operationCounter.countRequest("/inconnu");
    }

}

File: tp-arithmetic-kotlin/src/test/kotlin/fr/isima/service/OperationCounterTest.kt
package fr.isima.service


import org.junit.Test

import org.assertj.core.api.Assertions.assertThat

class OperationCounterTest {

    private val operationCounter = OperationCounter()

    @Test
    @Throws(Exception::class)
    fun countRequest_maths_returnsToStringWithMathIncrementedByOne() {
        assertThat(operationCounter.countRequest("/maths")).isEqualTo("Math ops : 1 and string ops 0")
    }

    @Test
    @Throws(Exception::class)
    fun countRequest_strings_returnsToStringWithStringsIncrementedByOne() {
        assertThat(operationCounter.countRequest("/strings")).isEqualTo("Math ops : 0 and string ops 1")
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(Exception::class)
    fun countRequest_noBoundedPath_thorws() {
        operationCounter.countRequest("/inconnu")
    }

}